tofhir-server/src/main/scala/io/tofhir/server/endpoint/ToFhirServerEndpoint.scala

```diff
@@ -46,8 +48,10 @@ class ToFhirServerEndpoint(toFhirEngineConfig: ToFhirEngineConfig, webServerConf
               optionalHeaderValueByName("X-Correlation-Id") { correlationId =>
                 val restCall = new ToFhirRestCall(method = httpMethod, uri = requestUri, requestId = correlationId.getOrElse(UUID.randomUUID().toString), requestEntity = requestEntity)
                 handleRejections(RejectionHandler.default) { // Default rejection handling
```

Comment:
We can get rid of the default rejection handler and add a generic handler to our own handler (CustomRejectionHandler) catching any unexpected error and still converting it a proper error format

Reply:
I added a generic handler but it does not send a descriptive message to the front-end. It only sends the names of the rejections encountered. Because AKKA rejections don't have a general descriptive error message. 
Reference: https://doc.akka.io/api/akka-http/10.6.0/akka/http/javadsl/server/Rejection.html